src/NuGet.Core/NuGet.Packaging/ContentModel/ContentPropertyDefinition.cs

```diff
@@ -15,76 +15,40 @@ public class ContentPropertyDefinition
     {
         private static readonly Func<object, object, bool> EqualsTest = (left, right) => Equals(left, right);
 
-        public ContentPropertyDefinition(string name)
-            : this(name, null, null, null, null, false)
-        {
-        }
-
-        public ContentPropertyDefinition(
+        internal ContentPropertyDefinition(
```

Comment:
Looks like all the constructors were changed to internal, plus a few properties.

Can the class itself be changed to internal, rather than just a few APIs?

Reply:
I honestly assumed that was already done :D 

It's exposed in ManagedCodeConventions, so I made that type of ManagedCodeConventions internal.

Reply:
Exposing the ContentPropertyDefinition type without constructors, pretty much locks you into whatever ManagedCodeConventions specified, so having that is seemingly not helpful, but it does allow you to reference specific property definitions like https://github.com/NuGet/NuGet.Client/blob/8e96cf5c2b5a192a3955b7e36951e482b2613915/src/NuGet.Core/NuGet.Packaging/Rules/UpholdBuildConventionRule.cs#L121-L132.

Reply:
Honestly, I can convince myself that theoretically it'd be helpful to have all types, but in practice, it's probably been made too difficult to customize this in any way shape or form. 

The ManagedCodeConventions type in particular is something that's used by others, so we can't really go too aggressive and make that internal. 

I can very much see why @jeffkl landed exactly here since it's a balance between allowing us to refactor that code easily in the future and minimize the chance of accidentally making the breaking changes so big, some people can't use the package anymore.

Reply:
Updated this comment. 

There's a NuGet.Commands usage that needs the type.